lib/benchee/benchmark/runner.ex

```diff
@@ -37,9 +37,17 @@ defmodule Benchee.Benchmark.Runner do
            config: config
          }) do
     printer.benchmarking(job_name, input_name, config)
-    Parallel.map(1..config.parallel, fn(_task_number) ->
+    1..config.parallel
+    |> Parallel.map(fn(_task_number) ->
       run_scenario(scenario, scenario_context)
     end)
+    |> Enum.group_by(fn scenario -> {scenario.function, scenario.input} end)
+    |> Enum.map(fn {_, like_results} ->
+      consolidated_results =
+        Enum.flat_map(like_results, fn scenario -> scenario.run_times end)
+      [scenario | _] = like_results
+      %Benchee.Benchmark.Scenario{scenario | run_times: consolidated_results}
+    end)
```

Comment:
I'm just thinking out loud here - this all seems quite a bit involved to get the results straight. What if we moved the point where we parallelized? I.e. a scenario is aware it's being run in parallel so it's already the scenario that just starts the function that reports back the `run_times` and concatenates them/flat_maps them whatever. Seems easier to me albeit involving moving some code around.

What do you think?

Reply:
I hadn't thought of that. I think it might be a good idea, but let me play around with it tomorrow and see what pops up. This solution did feel a little hack-y to me, but I couldn't come up with anything better at the time 😉